indigo_app/static/javascript/indigo/views/document_editor.js

```diff
@@ -667,6 +672,40 @@
       this.textEditor.focus();
     },
 
+    getAnnotationAmendingWork: function(document) {
+      var date = document.get('expression_date'),
+          documentAmendments = document.get('amendments'),
+          amendments = [];
+
+      documentAmendments.forEach(function(amendment) {
+        if (amendment.date === date) {
+          amendments.push(amendment);
+        }
+      });
+
+      if (amendments) {
+        return amendments[0];
+      }
+
+    },
+
+    insertAnnotation: function(e) {
+      e.preventDefault();
+      var element = 'section/subsection/paragraph/subparagraph/<other> ()',
+          verb = 'substituted/amended/inserted/deleted/repealed/<other>',
```

Comment:
What if the menu had a sub-menu that let the user choose the verb?

Reply:
Nice! I initially thought a pop-up form where you choose a verb and give it the other details as a second-generation version, but this could work nicely too. So we'd store a list of verbs per language somewhere and show the relevant language's list as sub options?

Reply:
I don't think it needs to be translated. Think of the menu as the user showing their intent: "I want to generate a remark about an insertion...". We then generate an appropriate, localised, remark.